sycl/source/detail/queue_impl.cpp

```diff
@@ -293,50 +285,31 @@ void queue_impl::wait(const detail::code_location &CodeLoc) {
   // directly. Otherwise, only wait for unenqueued or host task events, starting
```

Comment:
Can we move mutex on line 279 so it locks the whole functions to avoid races between multiple calls to `event::wait`?
Probably, it's too much for this PR though.

Reply:
We definitely should, but I would prefer to do this separately.